src/test/python/deep_qa_test/layers/backend/masked_batch_dot_test.py

```diff
@@ -0,0 +1,36 @@
+# pylint: disable=no-self-use,invalid-name
+import numpy
+from numpy.testing import assert_almost_equal
+from keras.layers import Input
+from keras.models import Model
+
+from deep_qa.layers.backend.masked_batch_dot import MaskedBatchDot
+
+class TestMaskedBatchDotLayer:
+    def test_call_works_on_simple_input(self):
+        batch_size = 2
+        tensor_a = Input(shape=(3, 2), dtype='float32')
+        tensor_b = Input(shape=(4, 2), dtype='float32')
+        masked_batch_dot = MaskedBatchDot()([tensor_a, tensor_b])
+        model = Model(input=[tensor_a, tensor_b], output=[masked_batch_dot])
+
+        tensor_a = numpy.array([[[0.3, 0.1], [0.4, 0.2], [0.8, 0.1]],
+                                [[0.3, 0.1], [0.4, 0.2], [0.8, 0.1]]])
+        tensor_b = numpy.array([[[0.2, 0.6], [0.4, 0.3], [0.5, 0.7], [0.1, .6]],
+                                [[0.2, 0.6], [0.4, 0.3], [0.5, 0.7], [0.1, .6]]])
+        adotb = model.predict([tensor_a, tensor_b])
+        assert adotb.shape == (batch_size, 3, 4)
+        assert_almost_equal(adotb, numpy.array([[[ 0.12, 0.15000001, 0.22, 0.09],
+                                                 [ 0.20000002, 0.22000001, 0.34, 0.16000001],
+                                                 [ 0.22000001, 0.35000002, 0.47, 0.14000002]],
+                                                [[ 0.12, 0.15000001, 0.22, 0.09],
+                                                 [ 0.20000002, 0.22000001, 0.34, 0.16000001],
+                                                 [ 0.22000001, 0.35000002, 0.47, 0.14000002]]]))
+
+    def test_get_output_shape_for(self):
+        # TODO(nelson): verify the correctness of this, esp. in rank 1 edge case.
```

Comment:
I based this off of the docs for `K.batch_dot`, and they had a special case when the final output is rank one but I'm not too sure if i captured it.